core/src/main/scala/com/iheart/happy/path/FutureEither.scala

```diff
@@ -112,14 +112,19 @@ private[path] abstract class FutureEitherFunctions {
   ): FutureEither[T] =
     optionToEitherReason[T](o, reason)
 
+  @deprecated("This one generates a RegularReason on the left side. Use the `ofEither[L, T](either: Either[L, T], ifLeft: L => Reason)` instead. ", "0.7.4")
   def ofEither[L, T](either: Either[L, T]): FutureEither[T] = eitherToEitherOrReason(either)
 
+  def ofEither[L, T](either: Either[L, T], ifLeft: L ⇒ Reason): FutureEither[T] = eitherToReason(either, ifLeft)
```

Comment:
a bit late, but `ifLeft` implies a boolean to me.  For `=> Reason`, may be `whenLeft` is more descriptive